sd_dynamic_prompts/dynamic_prompting.py

```diff
@@ -102,7 +102,20 @@ def generate_prompts(
     return all_prompts, all_negative_prompts
 
 
+already_loaded = False
+
+
 class Script(scripts.Script):
+    def __init__(self):
+        global already_loaded
+        if not already_loaded:
+            already_loaded = True
+            self._pnginfo_saver = PngInfoSaver()
+            self._promptwriter = PromptWriter()
```

Comment:
Small nit: Maybe `_png_info_saver` and `_prompt_writer`?

Reply:
I think `_pnginfo_saver` and `_prompt_writer` - `pnginfo` is what auto1111 uses, e.g. `image_save_params.pnginfo["parameters"]`

Reply:
Okiedoke - then maybe name the class `PnginfoSaver` for symmetry?